src/index.cpp

```diff
@@ -362,6 +375,10 @@ void Index<T, TagT, LabelT>::save(const char *filename, bool compact_before_save
                 assert(label_writer.is_open());
                 for (uint32_t i = 0; i < _pts_to_labels.size(); i++)
                 {
+                    if (_pts_to_labels[i].empty())
```

Comment:
what is this for?

Reply:
essentially, _pts_to_labels will have size num_pts  + num_frozen_pts, right? if so , see the comment in load section

Reply:
Yes, you are right, but in a case where total points is 12000 and we have max_pts_to_load is 10000, as we have in test_insert_delete_consolidate, the final points will have _pts_to_labels as empty. Notice the for loop is acting on **_pts_to_labels.size()** and not **_nd** which is number of active points.

Reply:
but wont this skip lines, if an intermediate point is deleted? we should at least write out a blank line right?

Reply:
I agree. How would we avoid the overflow condition.